### PR TITLE
Milestone 2 - Bookstore App

### PR DESCRIPTION
**Pull request by teekaytech**
### In this milestone, we initialized the Redux actions, reducers, and store. Additionally, we also defined empty React components, and we separated presentation and container components.

We added two container components (those which connect with the Redux store): BooksList and BooksForm and two presentational components: App and Book.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello Team 👋

**You've both done well on this.**👍

However, let's make a few changes and we are good to go.

- [ ] Please check the comments in your code.

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

File: src/components/BooksList.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      <tbody>{bookList}</tbody>
+    </table>
+  );
+};
+
+BooksList.propTypes = {
+  books: PropTypes.arrayOf(PropTypes.object).isRequired,
+};
+
+const mapStateToProps = state => ({
```
Comment (mikenath223) [inline src/components/BooksList.js]: - [x] Kindly place the `BooksList.js` file under the `src/containers` directory as it's a container component.

File: src/components/BookForm.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      id: null,
+      title: '',
+      category: '',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
```
Comment (mikenath223) [inline src/components/BookForm.js]: - [x] Please place the `BookForm.js` file under the `src/containers` directory.

File: src/reducers/index.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { combineReducers } from 'redux';
+import booksReducer from './book';
+// import filter from './filter';
```
Comment (mikenath223) [inline src/reducers/index.js]: - [x] Kindly remove this commented out code.

File: src/components/Book.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      <td>{book.id}</td>
+      <td>{book.title}</td>
+      <td>{book.category}</td>
+    </tr>
+  );
+};
+
+Book.propTypes = {
+  book: PropTypes.objectOf(
+    PropTypes.oneOfType([
```
Comment (Torres-ssf) [inline src/components/Book.js]: - [ ] `oneOfType` is used to an object that could be one of many types. The `book` object it's always going to be the same for all books, therefore, we need to describe its type with a different approach here. There many good alternatives but one that fit our case the most it's `exact`.

```JS
Book.propTypes = {
  book: PropTypes.exact({
    id: PropTypes.number,
    title: PropTypes.string,
    category: PropTypes.string,
  }).isRequired,
};
```

You can learn more about Typechecking With PropTypes [here](https://reactjs.org/docs/typechecking-with-proptypes.html)

File: src/containers/BookForm.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      title: '',
+      category: '',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit = event => {
+    // eslint-disable-next-line react/prop-types
```
Comment (Torres-ssf) [inline src/containers/BookForm.js]: - [x] We can remove this Eslint disable line and add a type checking to the component:
```JS
BookForm.propTypes = {
  createBook: PropTypes.func.isRequired,
};
```

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Changes required :muscle: 
Hi there team :hand: 
Great job in this milestone and implementing the requested changes.
However, there still changes to make before moving on.

- [x] We need to make your repository more professional by adding relevant information about your project to the README file. This requirement is listed on the Microverse general requirements. You can use this template [here](https://github.com/microverseinc/readme-template) in your README file. And you can also read this article to learn more about the [Professional Github repository](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/)
Screenshot from 2020-10-29 11-45-22

Feel free to message me on slack if you have any questions :smile: 
Happy Coding :computer: 
**Sergio Torres**

**Review by Torres-ssf — APPROVED** [review]
### Milestone approved :+1: :clap: 
Hi team :hand: 
Great job implementing the requested changes :smile: 
Good luck with the next milestone.


Happy Coding :computer: 
**Sergio Torres**

**Pull request merged**